<launch>
  <arg name="USE_LASER" default="true" />
  <arg name="USE_SCREENPOINT" default="true" />
  <arg name="USE_VIEW" default="false" />
  <arg name="USE_HAND"  default="true" />
  <arg name="USE_IMAGE_TRANSPORT" default="false" />
  <arg name="USE_OCTOMAP" default="true" />
  <arg name="USE_BDI_ACTION" default="true" />

  <!-- hrpsys controller launch -->
  <include file="$(find hrpsys_gazebo)/launch/atlas_hrpsys_bringup.launch" />

  <!-- ROS nodes -->
  <!-- laser scanning -->
  <include if="$(arg USE_LASER)"
           file="$(find hrpsys_gazebo)/launch/atlas_laser.launch" />

  <!-- screen point -->
  <include if="$(arg USE_SCREENPOINT)"
           file="$(find hrpsys_gazebo)/launch/atlas_screenpoint.launch" >
    <arg name="USE_VIEW" value="$(arg USE_VIEW)" />
  </include>

  <!-- image transport -->
  <include if="$(arg USE_IMAGE_TRANSPORT)"
           file="$(find hrpsys_gazebo)/launch/atlas_imagetransport.launch" />

  <!-- octomap -->
  <include if="$(arg USE_OCTOMAP)"
           file="$(find hrpsys_gazebo)/launch/atlas_octomap.launch" >
    <arg name="USE_TILT_LASER" value="false" />
  </include>

  <!-- hand controller -->
  <include if="$(arg USE_HAND)"
           file="$(find hrpsys_gazebo)/launch/sandia_hand_controller.launch" />

  <!-- bdi_controller -->
  <node if="$(arg USE_BDI_ACTION)"
        pkg="atlas_msgs" type="actionlib_server" name="walking_actionlib_server" />

  <!-- tf for sensors -->
  <include file="$(find hrpsys_gazebo)/launch/atlas_sensor_tf.launch" />
</launch>
